pkg/api/controller.go

```diff
@@ -3391,6 +3391,9 @@ func (c *Controller) GetObject(w http.ResponseWriter, r *http.Request, repositor
 	lastModified := httputil.HeaderTimestamp(entry.CreationDate)
 	w.Header().Set("Last-Modified", lastModified)
 	w.Header().Set("Content-Type", entry.ContentType)
+	// for security, make sure the browser and any proxies en route don't cache the response
+	w.Header().Set("Cache-Control", "no-store, must-revalidate")
+	w.Header().Set("Expires", "0")
```

Comment:
1. Consider using "no-cache, no-store, must-revalidate" for Cache-Control
2. Additional two headers to consider:
  w.Header().Set("pragma", "no-cache")
  w.Header().Set("X-Content-Type-Options", "nosniff")
3. If FF also cache the HEAD response we should also do the same for `headObject`
4. We should consider having it in the middleware when a request to GET/HEAD is handled under the `/api` route. WDYT?

Reply:
1. If `no-store` is used, `no-cache` doesn't make much sense. `no-cache` controls reuse, but if it's not stored public or private, there's nothing to reuse. It made sense for older browsers, but these days I don't think it's needed if you have `no-store`
2. The Pragma header is deprecated, and since IE is officially dead, I don't believe it's supported in any modern browser. Re `X-Content-Type-Options` - we usually return `octet-stream`, so we should allow browsers to sniff MIME type, if they want. WDYT?
3. I thought about it, but I'm not sure that's needed. The potentially sensitive content is in the GET, not the HEAD. WDYT?
4. Interesting... I like the idea, but are we sure it won't break anything?

Reply:
1. OK
2. Think you can leave it - I think for this request we return the associated content type - but not for this issue
3. Suggested to do the same for HEAD as the result is cacheable.
4. Can keep just the two for now - thanks!

Reply:
Added cache headers to HEAD as well